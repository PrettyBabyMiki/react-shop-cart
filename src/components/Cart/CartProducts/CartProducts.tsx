import { ICartContext, useCartContext } from 'contexts/cart-context';

import CartProduct from './CartProduct';
import * as S from './style';

const CartProducts = () => {
  const { products } = useCartContext() as ICartContext;

  return (
    <S.Container>
      {products?.length ? (
        products.map((p) => <CartProduct product={p} key={p.sku} />)
      ) : (
        <S.CartProductsEmpty>
          Add some products in the cart <br />
          :)
        </S.CartProductsEmpty>
      )}
    </S.Container>
  );
};

export default CartProducts;
