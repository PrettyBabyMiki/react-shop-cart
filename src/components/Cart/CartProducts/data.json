{
  "cart": {
    "products": [
      {
        "availableSizes": ["M"],
        "currencyFormat": "$",
        "currencyId": "USD",
        "description": "",
        "id": 13,
        "installments": 5,
        "isFreeShipping": true,
        "price": 29.45,
        "sku": 51498472915966370,
        "style": "Front print and paisley print",
        "title": "Dark Thug Blue-Navy T-Shirt",
        "quantity": 2
      },
      {
        "availableSizes": ["S", "XS"],
        "currencyFormat": "$",
        "currencyId": "USD",
        "description": "4 MSL",
        "id": 12,
        "installments": 9,
        "isFreeShipping": true,
        "price": 10.9,
        "sku": 12064273040195392,
        "style": "Black with custom print",
        "title": "Cat Tee Black T-Shirt",
        "quantity": 1
      }
    ],
    "productToAdd": {
      "availableSizes": ["S", "XS"],
      "currencyFormat": "$",
      "currencyId": "USD",
      "description": "4 MSL",
      "id": 12,
      "installments": 9,
      "isFreeShipping": true,
      "price": 10.9,
      "sku": 12064273040195392,
      "style": "Black with custom print",
      "title": "Cat Tee Black T-Shirt",
      "quantity": 1
    },
    "productToRemove": {
      "availableSizes": ["M"],
      "currencyFormat": "$",
      "currencyId": "USD",
      "description": "",
      "id": 13,
      "installments": 5,
      "isFreeShipping": true,
      "price": 29.45,
      "sku": 51498472915966370,
      "style": "Front print and paisley print",
      "title": "Dark Thug Blue-Navy T-Shirt",
      "quantity": 1
    },
    "productToChange": {
      "availableSizes": ["M"],
      "currencyFormat": "$",
      "currencyId": "USD",
      "description": "",
      "id": 13,
      "installments": 5,
      "isFreeShipping": true,
      "price": 29.45,
      "sku": 51498472915966370,
      "style": "Front print and paisley print",
      "title": "Dark Thug Blue-Navy T-Shirt",
      "quantity": 2
    }
  },
  "total": {
    "data": {
      "productQuantity": 3,
      "installments": 9,
      "totalPrice": 69.8,
      "currencyId": "USD",
      "currencyFormat": "$"
    }
  }
}
