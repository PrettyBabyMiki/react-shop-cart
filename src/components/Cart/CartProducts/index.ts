import CartProducts from './CartProducts';

export default CartProducts;
