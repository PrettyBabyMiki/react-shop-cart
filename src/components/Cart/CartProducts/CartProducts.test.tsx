import { renderWithThemeProvider } from 'utils/test/test-utils';
import { mockCartProducts } from 'utils/test/mocks';

import CartProducts from '.';

describe('[components] - CartProducts', () => {
  const setup = () => {
    return renderWithThemeProvider(
      <CartProducts products={mockCartProducts} />
    );
  };

  test('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
