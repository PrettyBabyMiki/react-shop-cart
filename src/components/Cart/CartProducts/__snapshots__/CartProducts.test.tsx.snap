// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[components] - CartProducts should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-furwcr cLrkbH"
      >
        <div
          class="sc-bdvvtL jueDJs"
        >
          <button
            class="sc-jRQBWg hOcWzm"
            title="remove product from cart"
          />
          <img
            alt="Cropped Stay Groovy off white"
            class="sc-iCfMLu kdysBI"
            src="8552515751438644-1-cart.webp"
          />
          <div
            class="sc-gsDKAQ ixJPSE"
          >
            <p
              class="sc-dkPtRN gXtprY"
            >
              Cropped Stay Groovy off white
            </p>
            <p
              class="sc-hKwDye bPeUpS"
            >
              S | White T-shirt
               
              <br />
              Quantity: 
              1
            </p>
          </div>
          <div
            class="sc-eCImPb bHNFYi"
          >
            <p>
              $  10.90
            </p>
            <div>
              <button
                class="sc-gKclnd fUHPHw"
                disabled=""
              >
                -
              </button>
              <button
                class="sc-gKclnd fUHPHw"
              >
                +
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-bdvvtL jueDJs"
        >
          <button
            class="sc-jRQBWg hOcWzm"
            title="remove product from cart"
          />
          <img
            alt="Basic Cactus White T-shirt"
            class="sc-iCfMLu kdysBI"
            src="39876704341265610-1-cart.webp"
          />
          <div
            class="sc-gsDKAQ ixJPSE"
          >
            <p
              class="sc-dkPtRN gXtprY"
            >
              Basic Cactus White T-shirt
            </p>
            <p
              class="sc-hKwDye bPeUpS"
            >
              X | Wine
               
              <br />
              Quantity: 
              1
            </p>
          </div>
          <div
            class="sc-eCImPb bHNFYi"
          >
            <p>
              $  13.25
            </p>
            <div>
              <button
                class="sc-gKclnd fUHPHw"
                disabled=""
              >
                -
              </button>
              <button
                class="sc-gKclnd fUHPHw"
              >
                +
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-bdvvtL jueDJs"
        >
          <button
            class="sc-jRQBWg hOcWzm"
            title="remove product from cart"
          />
          <img
            alt="Skater Black Sweatshirt"
            class="sc-iCfMLu kdysBI"
            src="9197907543445676-1-cart.webp"
          />
          <div
            class="sc-gsDKAQ ixJPSE"
          >
            <p
              class="sc-dkPtRN gXtprY"
            >
              Skater Black Sweatshirt
            </p>
            <p
              class="sc-hKwDye bPeUpS"
            >
              XL | Tony Hawk
               
              <br />
              Quantity: 
              1
            </p>
          </div>
          <div
            class="sc-eCImPb bHNFYi"
          >
            <p>
              $  25.90
            </p>
            <div>
              <button
                class="sc-gKclnd fUHPHw"
                disabled=""
              >
                -
              </button>
              <button
                class="sc-gKclnd fUHPHw"
              >
                +
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-furwcr cLrkbH"
    >
      <div
        class="sc-bdvvtL jueDJs"
      >
        <button
          class="sc-jRQBWg hOcWzm"
          title="remove product from cart"
        />
        <img
          alt="Cropped Stay Groovy off white"
          class="sc-iCfMLu kdysBI"
          src="8552515751438644-1-cart.webp"
        />
        <div
          class="sc-gsDKAQ ixJPSE"
        >
          <p
            class="sc-dkPtRN gXtprY"
          >
            Cropped Stay Groovy off white
          </p>
          <p
            class="sc-hKwDye bPeUpS"
          >
            S | White T-shirt
             
            <br />
            Quantity: 
            1
          </p>
        </div>
        <div
          class="sc-eCImPb bHNFYi"
        >
          <p>
            $  10.90
          </p>
          <div>
            <button
              class="sc-gKclnd fUHPHw"
              disabled=""
            >
              -
            </button>
            <button
              class="sc-gKclnd fUHPHw"
            >
              +
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-bdvvtL jueDJs"
      >
        <button
          class="sc-jRQBWg hOcWzm"
          title="remove product from cart"
        />
        <img
          alt="Basic Cactus White T-shirt"
          class="sc-iCfMLu kdysBI"
          src="39876704341265610-1-cart.webp"
        />
        <div
          class="sc-gsDKAQ ixJPSE"
        >
          <p
            class="sc-dkPtRN gXtprY"
          >
            Basic Cactus White T-shirt
          </p>
          <p
            class="sc-hKwDye bPeUpS"
          >
            X | Wine
             
            <br />
            Quantity: 
            1
          </p>
        </div>
        <div
          class="sc-eCImPb bHNFYi"
        >
          <p>
            $  13.25
          </p>
          <div>
            <button
              class="sc-gKclnd fUHPHw"
              disabled=""
            >
              -
            </button>
            <button
              class="sc-gKclnd fUHPHw"
            >
              +
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-bdvvtL jueDJs"
      >
        <button
          class="sc-jRQBWg hOcWzm"
          title="remove product from cart"
        />
        <img
          alt="Skater Black Sweatshirt"
          class="sc-iCfMLu kdysBI"
          src="9197907543445676-1-cart.webp"
        />
        <div
          class="sc-gsDKAQ ixJPSE"
        >
          <p
            class="sc-dkPtRN gXtprY"
          >
            Skater Black Sweatshirt
          </p>
          <p
            class="sc-hKwDye bPeUpS"
          >
            XL | Tony Hawk
             
            <br />
            Quantity: 
            1
          </p>
        </div>
        <div
          class="sc-eCImPb bHNFYi"
        >
          <p>
            $  25.90
          </p>
          <div>
            <button
              class="sc-gKclnd fUHPHw"
              disabled=""
            >
              -
            </button>
            <button
              class="sc-gKclnd fUHPHw"
            >
              +
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
